sandbox/lua/decoders/json.lua

```diff
@@ -0,0 +1,170 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+Parses a payload containing JSON.
+
+Config:
+
+- type (string, optional, default "json"):
+    Sets the message 'Type' header to the specified value,
+    will be overridden if Type config option is specified.
+
+- payload_keep (bool, optional, default false)
+    Whether to preserve the original log line in the message payload.
+
+- map_fields (bool, optional, default false)
+    Enables mapping of json fields to heka message fields.
+
+- Uuid (string, optional, default nil)
+    string specifying json field to map to message Uuid,
+    expects field value to be a string.
+
+- Type (string, optional, default nil)
+    string specifying json field to map to to message Type,
+    expects field value to be a string.
+
+- Logger (string, optional, default nil)
+    string specifying json field to map to message Logger,
+    expects field value to be a string.
+
+- Hostname (string, optional, default nil)
+    string specifying json field to map to message Hostname,
+    expects field value to be a string.
+
+- Severity (string, optional, default nil)
+    string specifying json field to map to message Severity,
+    expects field value to be numeric.
+
+- EnvVersion (string, optional, default nil)
+    string specifying json field to map to message EnvVersion,
+    expects field value to be numeric.
+
+- Pid (string, optional, default nil)
+    string specifying json field to map to message Pid,
+    expects field value to be numeric
+
+- Timestamp (string, optional, default nil)
+    string specifying json field to map to message Timestamp,
+    expects field value to be unix epoch timestamp in seconds.
+
```

Comment:
Might be nice to support `Payload`, which would override `payload_keep` like `Type` overrides `type`.